File: Cpp/Score of Parentheses.cpp
// 0ms, 100.0%
static int x = []() {
    ios_base::sync_with_stdio(false);
    cin.tie(nullptr);
    return 0;
}();

class Solution
{
  public:
    int scoreOfParentheses(const string &str)
    {
        int res = 0, layer = 0;
        for (int i = 0, size = str.size() - 1; i < size; ++i) {
            layer += (str[i] == '(' ? 1 : -1);
            if (str[i] == '(' && str[i + 1] == ')') res += 1 << (layer - 1);
        }
        return res;
    }
};

File: Python3/Score of Parentheses.py
# 36ms, 97.67%
class Solution:
    def scoreOfParentheses(self, S):
        """
        :type S: str
        :rtype: int
        """
        res = layer = 0
        for i in range(0, len(S) - 1):
            if S[i] == '(':
                layer += 1
                if S[i + 1] == ')': res += 1 << (layer - 1)
            else:
                layer -= 1
        return res
